### ISSUE TITLE
bffnt incorrectly decodes ETC1 image

### ISSUE THREAD

**Issue by rawlinson** [ObsidianX/3dstools#17]
I have a font file for which bffnt fails to produce valid pngs. The resulting images contain many 8x8 blocks as expected, but the colours are all incorrect. Incidentally, I have to force the version to be 4 in code to allow it parse successfully. Font file attached.
[turbo_messageFontOutline_50.zip](https://github.com/ObsidianX/3dstools/files/1366451/turbo_messageFontOutline_50.zip)

**Comment by Tyulis** [comment]
This is known... And we search how to fix (see #10). Thanks for the file, concerned files to work on are always helpful.

**Comment by Tyulis** [comment]
I'll pull request for version issue

**Comment by Tyulis** [comment]
I tried to manually extract BFLIM sections of the file, and convert them with bflim.py, got the same result. 2 possibilities: there is a problem with the decompression algorithm (not probable, all ETC1 BFLIM files wirk fine), or it is not ETC1...

**Comment by rawlinson** [comment]
It turns out it is actually compressed as ATI1N/BC4U. So the pixel format specified in the TGLP header doesn't seem to be enough to identify the actual format used. Also, as I said, I had to force the version to 4 to use the correct headers despite the version being specified as 3.

**Comment by Tyulis** [comment]
I pull requested for the version issue, waiting for merge. The pixel format specified in TGLP works for any uncompressed font I tested, so ETC1 is probably replaced with another compression. I will try to work on the compressions you said. Thanks for help!

**Comment by Tyulis** [comment]
You're right, after searching in BFFNTtoGTX and on the web, I found that it's compressed in BC4.

**Comment by rawlinson** [comment]
Your fix for the version issue has already been merged and was used to open this file. It reports itself as version 3 but only works with the version 4 headers... and even then not perfectly, I had to modify the TGLP header struct to be "%s4sHxx4BI6HI", note the "Hxx" instead of "I".

It is a GameCube file so perhaps the formats aren't consistent between systems. If GameCube support is wanted then I think a system switch on the command line will be needed.

**Comment by Tyulis** [comment]
No, my previous version fix introduced some confusions between BCFNTv3 and BFFNTv3, I made another PR to split into BCFNT and BFFNT to avoid this, and I wait for merge. The problem has been found on some WiiU files, so i don't think this is specific on GameCube. I have to see if there are structure problems (as you said, v4 structures work, and appear to unpack coherent data), and for texture conversion.
